sklearn/metrics/classification.py

```diff
@@ -892,6 +903,12 @@ def f1_score(y_true, y_pred, labels=None, pos_label=1, average='binary',
     sample_weight : array-like of shape = [n_samples], optional
         Sample weights.
 
+    zero_division : string or int, default="warn"
```

Comment:
```suggestion
    zero_division : "warn", 0 or 1
```

Hmm... thinking about it, f1_score will be 0 in all cases. So this is only about suppressing the warning, if we're interested in such API-level control.

Reply:
Please give me some time (I'll do it during the weekend) to review and do another commit. I just merged a very conflicting PR and this needs changes. Regarding the values of f1, we agreed that when y_true and y_pred have none of them positives, we would give the user the option to return 1. Why now this change?